src/boot/bootctl-status.c

```diff
@@ -639,6 +639,65 @@ static int purge_entry(const BootConfig *config, const char *fn)
         return 0;
 }
 
+static int list_remove_orphaned_file(
+                RecurseDirEvent event,
+                const char *path,
+                int dir_fd,
+                int inode_fd,
+                const struct dirent *de,
+                const struct statx *sx,
+                void *userdata) {
+
+        Hashmap* known_files = userdata;
+
+        assert(path);
+        assert(known_files);
+
+        if (event == RECURSE_DIR_ENTRY) {
+                if (hashmap_get(known_files, path) == NULL) {
+                        log_info("removing orphaned file %s\n", path);
+                        int r = unlink(path);
+                        if (r < 0)
+                                log_error_errno(r, "failed to remove \"%s\": %m", path);
```

Comment:
unlink() returns error in errno hence first param must be `errno`, not `r`

Reply:
also, please downgrade to log_warning_errno() and suffix the message with ", ignoring".

we have this rule that errors that we skip over but still are relevant should never be logged above LOG_WARN log level, and that we should say so by suffixing the message with ", ignoring", to make clear what is happening.